File: 148-sort-list/148-sort-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var sortList = function(head) {
    const arr = [];
    let curr = head;
    while(curr!==null){
        arr.push(curr.val);
          curr=curr.next;
    }
    
    arr.sort((a,b)=> a-b);
    let res = new ListNode(0, null);
    let tail = res;
    for(let num of arr){
        tail.next = new ListNode(num, null);
        tail = tail.next;
    }
    
    return res.next;
};

File: 148-sort-list/148-sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode sortList(ListNode head) {
        ArrayList<Integer> arr = new ArrayList<>();
        ListNode curr = head;
        while(curr!=null){
            arr.add(curr.val);
            curr=curr.next;
        }

       Collections.sort(arr);
        ListNode res = new ListNode(0, null);
        ListNode tail = res;
        for(int i=0; i<arr.size(); i++){
            tail.next = new ListNode(arr.get(i), null);
            tail = tail.next;
        }

        return res.next;
        }
}